File: C++/idade.cpp
#include <iostream>
#include <iomanip>
#include <string>
#include <climits>

using namespace std;

int main(){

    int idade1, idade2;
    double media;
    string nome1, nome2;

    cout << "digite o nome da primeira pessoa: " << endl;
    getline(cin, nome1);
    cout << "digite a idade da primeira pessoa: " << endl;
    cin>>idade1;
    cout << "digite o nome da segunda pessoa: " << endl;
    cin.ignore(INT_MAX, '\n');
    getline(cin, nome2);
    cout << "digite a idade da segunda pessoa: " << endl;
    cin>>idade2;

    media = (double)(idade1 + idade2) /(double) 2;
    cout << fixed << setprecision(1);
    cout << "dados da primeira pessoa" << endl;
    cout << "nome:" << nome1 << endl;
    cout << "idade:" << idade1 << endl;
    cout << "dados da segunda pessoa" << endl;
    cout << "nome:" << nome2 << endl;
    cout << "idade:" << idade2 << endl;
    cout << "a media de idade eh de " << media << " anos" << endl;


    return 0;
}

File: Java/idade.java
import java.util.Scanner;
import java.util.Locale;

public class idade {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Locale.setDefault(Locale.US);

        String nome1, nome2;
        int idade1, idade2; 
        double media;

        System.out.print("digite o nome da primeira pessoa: ");
        nome1 = sc.nextLine();
        System.out.print("digite a idade da primeira pessoa:");
        idade1 = sc.nextInt();
        sc.nextLine();
        System.out.print("digite o nome da segunda pessoa: ");
        nome2 = sc.nextLine();
        System.out.print("digite a idade da segunda pessoa:");
        idade2 = sc.nextInt();

        media = (double)(idade1 + idade2)/(double)2;

        System.out.println("Dados da primeira pessoa:"
        +"\nNome: "+nome1+
        "\nIdade: "+idade1+
        "\nDados da segunda pessoa:"+
        "\nNome: "+nome2+
        "\nIdade: "+idade2+
        "\nA media de idade eh de "+media);
        
        sc.close();
    }
}
